pages/en/lb4/Dependency-Injection.md

```diff
@@ -43,10 +43,16 @@ In LoopBack, we use [Context](Context.html) to keep track of all injectable depe
 
 There are several different ways for configuring the values to inject, the simplest options is to call `app.bind(key).to(value)`. Building on top of the example above, one can configure the app to use a Basic HTTP authentication strategy as follows:
 
-```js
+```ts
+// TypeScript example
+
 import {BasicStrategy} from 'passport-http';
+import {Application} from '@loopback/core';
+import {RestServer} from '@loopback/rest';
+// basic scaffolding stuff happens in between...
 
-app.bind('authentication.strategy').to(new BasicStrategy(loginUser));
+const server = app.getServer(RestServer); // The REST server has its own context!
+server.bind('authentication.strategy').to(new BasicStrategy(loginUser));
```

Comment:
I'm not sure how we would even address something like this. We could easily have the examples use application-level context to do their work, since it's all available at the child context of the server, but then we're potentially encouraging a pattern that might prevent use of separate strategies in multiple server instances.

If we're going to alter the examples to simplify the bindings they work with, we'll need to make a section somewhere that details more advanced usage scenarios so that users properly separate their contextual bindings.